tests/infra_test.go

```diff
@@ -996,6 +999,251 @@ var _ = Describe("[Serial]Infrastructure", func() {
 		})
 
 	})
+
+	Describe("Node-labeller", func() {
+		Context("basic labelling", func() {
+			var nodesWithKVM []*k8sv1.Node
+
+			BeforeEach(func() {
+				tests.BeforeTestCleanup()
+				nodesWithKVM = tests.GetNodesWithKVM()
+			})
+
+			It("label nodes with cpu model and cpu features", func() {
+				for _, node := range nodesWithKVM {
+					Expect(err).ToNot(HaveOccurred())
+					cpuModelLabelPresent := false
+					cpuFeatureLabelPresent := false
+					hyperVLabelPresent := false
+					for key := range node.Labels {
+						if strings.Contains(key, services.NFD_CPU_MODEL_PREFIX) {
+							cpuModelLabelPresent = true
+						}
+						if strings.Contains(key, services.NFD_CPU_FEATURE_PREFIX) {
+							cpuFeatureLabelPresent = true
+						}
+						if strings.Contains(key, services.NFD_KVM_INFO_PREFIX) {
+							hyperVLabelPresent = true
+						}
+
+						if cpuModelLabelPresent && cpuFeatureLabelPresent && hyperVLabelPresent {
+							break
+						}
+					}
+					Expect(cpuModelLabelPresent).To(Equal(true), "node "+node.Name+" doesn't contain cpu label")
+					Expect(cpuFeatureLabelPresent).To(Equal(true), "node "+node.Name+" doesn't contain feature label")
+					Expect(hyperVLabelPresent).To(Equal(true), "node "+node.Name+" doesn't contain hyperV label")
+				}
+			})
+
+			It("should set default obsolete cpu filter when obsolete-cpus is not set in kubevirt config", func() {
+				node := nodesWithKVM[0]
+
+				for key := range node.Labels {
+					if strings.Contains(key, services.NFD_CPU_MODEL_PREFIX) {
+						model := strings.TrimPrefix(key, services.NFD_CPU_MODEL_PREFIX)
+						for _, obsoleteModel := range nodelabeller.DefaultObsoleteCPUs {
+							Expect(model == obsoleteModel).To(Equal(false), "Node can't contain label with cpu model, which is in default obsolete filter")
+						}
+					}
+				}
+			})
+
+			It("should set default min cpu filter when min-cpu is not set in kubevirt config", func() {
```

Comment:
`min cpu model` - I keep pointing this out because later it will cause confusion.